draft-ietf-quic-qlog-main-schema.md

```diff
@@ -949,103 +930,114 @@ fields are "time" and "data", who are divergent by nature.
 
-Concrete event types belong to event categories, both contained in event schema.
-
-A single event schema can either define a new namespace, or extend an existing
-namespace with new categories. New namespaces MUST be registered using a
-non-empty namespace identifier text identifier using only characters in the
-unreserved range; see {{Section 2.3 of RFC3986}}. Namespace are mutable and MAY
-be extended with categories.
-
-Event categories MUST belong to a single event namespace. They MUST have a
-registered non-empty globally-unique text identifier using only characters in
-the URI unreserved range; see {{Section 2.3 of RFC3986}}. They MUST have a
-single URI {{RFC3986}} that MUST be absolute. The URI MUST include the namespace
-identifier. The URI MUST include the category identifer using a fragment
-identifier (characters after a "#" in the URI). Event categories are immutable
-and MUST NOT be extended with events. Registration guidance and requirement for
-category URIs are provided in {{event-schema-reg}}.
-
-Concrete event types MUST belong to a single event category and MUST have a
-non-empty name of type `text`.
-
-The value of a qlog event `name` field MUST be the concatenation of category
-identifier, colon (':'), and event type identifier. By virtue of the identifier
-requirements described above, event names are globally-unique. Thus, log files
-can contain events from multiple schemas without the risk of name collisions.
-
-Implementations that might record concrete event types belonging to a category
-in an event schema SHOULD list all category identifiers in use. This is achieved
-by including the appropriate URIs in the `event_schemas` field of the abstract
-LogFile class ({{abstract-logfile}}). The `event_schemas` is a hint to tools
-about the possible event categories (and event types contained therein) that a
-qlog file might contain. The file MAY contain event types that do not belong to
-a listed category identifier. Inversely, not all event types associated with a
-category listed in `event_schemas` are guaranteed to be logged in a qlog file.
-Tools MUST NOT treat either of these as an error; see {{tooling}}.
-
-In the following hypothetical example, a qlog file contains events belonging to
-two categories in the generic event schema defined in this document
-({{generic-event-schema}}), to three categories in an event schema named `rick`
-specified in a hypothetical RFC, and to three more categories from a private
-event schema named pickle. The standardized category URIs use a URN format, the
-private categories use a URI with domain name.
+Concrete event types (for example: packet_sent), as well as related data types
+(for example: QUICPacketHeader), are grouped in event namespaces (for example:
+quic) which in turn are defined in one or multiple event schemas (for example:
+urn:ietf:params:qlog:events:quic and
+urn:ietf:params:qlog:events:quic#extension1).
```

Comment:
nit: I find the inband examples break up the flow, so perhaps consider an example underneath